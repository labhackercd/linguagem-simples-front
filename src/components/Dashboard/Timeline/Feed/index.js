import React from 'react';
import {Paper,Button, TextField, Dialog, DialogActions, DialogContent, DialogContentText,
	      DialogTitle, Grid, Typography, Box,  List, ListItem} from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';
import {TwitterTweetEmbed} from 'react-twitter-embed';
import MoreOptionsIcon from './../../../../assets/more_options_icon.svg';
import { parseHourMinute } from './../../../Util';
import { ReactTinyLink } from 'react-tiny-link';

const useStyles = makeStyles((theme) => ({
  updatesArea: {
    overflow: 'auto',
    maxHeight: '50vh',
  },
  itemHeader: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  title: {
    color: theme.palette.primary.main
  },
  timelinePost: {
    overflow: 'auto',
    overflowWrap: 'break-word',
    display: 'flex',
    margin: '1rem 0',
    maxHeight: '25vh',
  },
}))

export default function Feed(props) {
  const classes = useStyles();
  return (
    <Grid container className={classes.updatesArea}>
      <List style={{width: '100%'}}>
          {props.updates ? props.updates.slice(0).reverse().map(update =>  //creates a shallow copy of the array and reverses it
						<ListItem divider key={update.id}>
              <Grid container>
                <Grid item xs={2}>
                  <Typography variant="h6">
										{parseHourMinute(update.created)}
                  </Typography>
                </Grid>
                <Grid item xs={10}>
                  <Grid container className={classes.itemHeader}>
                    <Typography variant="h5" className={classes.title}>
                      {update.title}
                    </Typography>
                    <Grid item>
                      <img src={MoreOptionsIcon} alt="more options icon"/>
                    </Grid>
                  </Grid>
                  <Grid container className={classes.timelinePost}>
                    <div style={{width: '100%'}}>
                      <Typography style={{color: '#666', alignSelf: 'flex-start'}} variant="body1">
												{update.updateTextArea ?
												 update.updateTextArea :
												 ''
												}</Typography>
                    </div>
                    <section style={{width: '100%'}}>
                      {update.tweet_id ?
                        <TwitterTweetEmbed
                          style={{width: '100%', margin: '0 0 1rem 0'}}
                          tweetId={update.tweet_id}
                        /> : '' }
                      {update.image ?
                       <img src={update.image}  style={{maxWidth: '100%', maxHeight: '30vh'}} alt="timeline post"/> : '' }
											{update.customURL && !(props.URLInputIsTwitter) ?
												<ReactTinyLink
													cardSize="small"
													showGraphic={true}
													maxLine={2}
													minLine={1}
													url={update.customURL} /> :
													''}
                    </section>
                  </Grid>
                </Grid>
              </Grid>
            </ListItem>
          ) : ''}
      </List>
    </Grid>
  )
}
