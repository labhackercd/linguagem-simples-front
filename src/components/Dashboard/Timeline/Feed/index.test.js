import React from 'react';
import { render, screen } from '@testing-library/react';
import Feed from './index';
import {shallow} from "enzyme/build";

it("should render the Feed section", () => {
    const props = {
      updates: [{
        id: 1,
        content: 'test',
        time: '18:00',
      }]
    }
    const component = shallow(<Feed {...props} />);
    expect(component).toMatchSnapshot();
});
