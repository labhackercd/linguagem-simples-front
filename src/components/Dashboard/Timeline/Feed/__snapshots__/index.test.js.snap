// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Feed section 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-updatesArea-1"
  container={true}
>
  <WithStyles(ForwardRef(List))
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <WithStyles(ForwardRef(ListItem))
      divider={true}
      key="1"
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={2}
        >
          <WithStyles(ForwardRef(Typography))
            variant="h6"
          >
            18:00
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={10}
        >
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-itemHeader-2"
            container={true}
          >
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-title-3"
              variant="h5"
            />
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <img
                alt="more options icon"
                src="../../img/more_options_icon.svg"
              />
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            className="makeStyles-timelinePost-4"
            container={true}
          >
            <div
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <WithStyles(ForwardRef(Typography))
                style={
                  Object {
                    "alignSelf": "flex-start",
                    "color": "#666",
                  }
                }
                variant="body1"
              >
                test
              </WithStyles(ForwardRef(Typography))>
            </div>
            <section
              style={
                Object {
                  "width": "100%",
                }
              }
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
</WithStyles(ForwardRef(Grid))>
`;
